src/Validator/Constraints/AttributeTypeForOptionValidator.php

```diff
@@ -30,11 +29,11 @@ public function validate($attributeOption, Constraint $constraint)
             $authorizedTypes = [
                 AttributeTypes::OPTION_SIMPLE_SELECT,
                 AttributeTypes::OPTION_MULTI_SELECT,
-                TableType::FLAGBIT_CATALOG_TABLE
+                TableType::FLAGBIT_CATALOG_TABLE,
             ];
 
             if (!in_array($attribute->getAttributeType(), $authorizedTypes)) {
```

Comment:
The parent class also checks `null !== $attribute` here. I'm not sure if this case can really happen, because the interface of `AttributeOption::getAttribute()` only lists `AttributeInterface` as a valid return type. Maybe Akeneo PIM is wrong there, but maybe Akeneo is not true to the contract of their own classes. What do you think about this state of Akeneo PIM?

Reply:
Since there is not a explicit contract on the interface defining the return type, I would keep the check `null !== $attribute`.